### PR TITLE
Extend test coverage to sync relationships

### PR DESCRIPTION
**Pull request by andreyfel**
This PR extends test coverage for sync and async relationships.

### CODE REVIEW

**Comment by snewcomer** [comment]
I like this!!  👍 Happen to know why the two tests are failing?

Also to confirm, there is no addon code changes needed right?

**Comment by andreyfel** [comment]
After recent changes added in #404 those 2 tests started failing. It is caused by the fact that `isEmberDataObject` util method doesn't work with sync relationships. We need a more reliable approach.
cc: @josemarluedke

**Comment by snewcomer** [comment]
Great points!  Yes you are right.  A couple of points.

1.  We need to add a check for `obj._internalModel`, in the EmberData check.  Seems that initial check is not comprehensive.
2.  We aren't checking if is a type of array in the referenced PR before doing Object.assign.  So it would have to look something like this...

`if (!Array.isArray(result) && isObject(content[baseKey]) && !isEmberDataObject(content[baseKey])) {`

**Comment by snewcomer** [comment]
I merged the Array check #413

**Comment by andreyfel** [comment]
_internalModel is not a public API, so, it can be removed at any moment. I've asked for advice in the ED discord channel.

**Comment by andreyfel** [comment]
@runspired suggested that we should ask the user if the object is an Ember Data Model or not.
So, we should either have something like a flag passed in the Changeset constructor which says if an object is an ED model or even a separate class for ED models.

**Comment by snewcomer** [comment]
@andreyfel I like the idea of explicitly saying it is an EmberData object.  However, this might get a little murky when an object's nested values are EmberData objects.

A few thoughts.

1.  We can bring back the `changeset-get` helper.  Ref #414.  Ultimately, requesting for the leaf key should actually ask for the leaf key and ideally the logic introduced in #404 shouldn't be necessary.
2.  However, based on `master`, it is necessary if the end user calls `changeset.get('user').lastName` when only `user.firstName` has changes because right now, `lastName` will be the `model` value.  I think there is a fundamental problem I have built into the proxied `get` that has led this discussion astray.  It is this line - https://github.com/poteto/ember-changeset/blob/master/addon/-private/validated-changeset.ts#L896.  `changeset.get('user').lastName` should return `undefined` and not the original value on the model.  

The combination of these two would alleviate all our problems, albeit, a stricter API.  Do you have any thoughts?

File: tests/integration/main-test.js [hunk 1]
```diff
@@ -171,5 +200,42 @@ module('Integration | main', function(hooks) {
...
+      this.store.createRecord('dog', { breed: 'Münsterländer' })
+    ];
+
+    changeset.set('dogs', dogs);
+
+    changeset.execute();
+
+    assert.equal(user.get('profile.firstName'), 'Bob', 'Profile is set on user');
+    assert.equal(user.get('dogs.firstObject.breed'), 'rough collie');
+    assert.equal(user.get('dogs.lastObject.breed'), 'Münsterländer');
```
Comment (snewcomer) [inline tests/integration/main-test.js]: FYI I don't think these array accessors work (at the moment at least)

Comment (andreyfel) [reply tests/integration/main-test.js]: Well, they work for EmberArray and `dogs` is an EmberArray.

Comment (snewcomer) [reply tests/integration/main-test.js]: Oh you are right...we are passing in Ember's getter...

**Comment by snewcomer** [comment]
@andreyfel Is this PR g2g?  I'm happy with it in its current state as long as the latest master is merged in...

**Comment by andreyfel** [comment]
Yes, I've rebased it today. The pr can be merged and we can think of refactoring after that.

**Comment by andreyfel** [comment]
@snewcomer yesterday I wrote a comment but it appears I forgot to publish it.
Regarding `object's nested values are EmberData objects`. Is this feature supported? Are there are tests for that scenario? Maybe we don't need to support that?
Thus PR is kind of a bandaid fix to make tests pass and merge them in but I believe we should do something about it in the long term.

**Pull request merged**